automation/Dockerfile

```diff
@@ -3,37 +3,20 @@
 # It enables systemd to be operational.
 FROM centos:7.5.1804
 ENV container docker
-RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == \
-     systemd-tmpfiles-setup.service ] || rm -f $i; done); \
-    rm -f /lib/systemd/system/multi-user.target.wants/*;\
-    rm -f /etc/systemd/system/*.wants/*;\
-    rm -f /lib/systemd/system/local-fs.target.wants/*; \
-    rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
-    rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
-    rm -f /lib/systemd/system/basic.target.wants/*;\
-    rm -f /lib/systemd/system/anaconda.target.wants/*;
-
-# Enable EPEL and centos-release-openstack-rocky repo for new version of pytest.
-RUN yum -y install epel-release centos-release-openstack-rocky
-
-# Enable nmstate test repo for new version of pytest-cov and NetworkManager bug
-# fix of https://bugzilla.redhat.com/show_bug.cgi?id=1642625
-RUN yum-config-manager --add-repo https://copr.fedorainfracloud.org/coprs/nmstate/nmstate-test-env-el/repo/epel-7/nmstate-nmstate-test-env-el-epel-7.repo && \
-    yum-config-manager --enable nmstate-nmstate-test-env-el
+COPY docker_enable_systemd.sh docker_sys_config.sh ./
+RUN bash ./docker_enable_systemd.sh && rm ./docker_enable_systemd.sh -f
 
 # libibverbs is a dependency for openvswitch that is missing as a dependency in the RPM
 RUN yum -y upgrade && \
-    yum -y install iproute python2-pytest python2-pytest-cov NetworkManager \
+    yum -y install epel-release && \
+    yum -y install python2-pip iproute NetworkManager \
                    rpm-build git python2-devel python2-six \
                    python2-pyyaml python-jsonschema python-setuptools \
                    NetworkManager-ovs openvswitch libibverbs \
                    python-gobject-base dnsmasq && \
+    pip install -U pip pytest==4.1.0 pytest-cov==2.6.1 python-coveralls && \
```

Comment:
Please pin `python-coveralls` as well.

BTW: The `-U` is not needed here, it is not an upgrade but a fresh install.

Reply:
> Please pin `python-coveralls` as well.

Is it a good idea since it is used to access an external service which is not pinned and it might also have security implications?

Reply:
The `-U` is required as pip itself should be upgrade before install others, that's what old code did.

Reply:
-U just tels pip to upgrade a package if it is already installed (it will not upgrade and leave it as is otherwise).

I would expect the service to support old versions. If you do not pin, it may not work with the current version of python in the future.

Reply:
So should pip be removed from the list to the installed pkgs? For consistency I can live with pinning coveralls as well.

Reply:
@EdDev you introduced the pip command-line initially with `-U pip ...` - so I guess there must have been a reason back then.